### ISSUE TITLE
legion spy issue

### ISSUE THREAD

**Issue by bandokihiro** [StanfordLegion/legion#1321]
While investigating an issue related to corrupted HDF5 outputs, I stumbled into this.

Config:

- legion 7d6c32e1f built with `-DLEGION_SPY`
- 2-rank run (1 rank per node)
- command line: `mpiexec -np 2 mbind.x -cs -t40 -v ../builds/base_1d/exec/solver -logfile logs/log_%.log -hdf5:forcerw -ll:cpu 0 -ll:util 2 -ll:csize 20000 -ll:rsize 0 -ll:onuma 0 -ll:ht_sharing 1 -ll:ocpu 1 -ll:othr 18 -lg:spy`

Processing the logs with `legion_spy -l log*` produced the following output:
```
INFO: Falling back to transitive mapping dependences for Inter Close Op 691 due to either imprecision in the runtime analysis or a bug.
INFO: Falling back to transitive mapping dependences for forward_euler_task due to either imprecision in the runtime analysis or a bug.
INFO: Falling back to transitive mapping dependences for Inter Close Op 697 due to either imprecision in the runtime analysis or a bug.
Warning: skipping logical analysis of Attach Op 715 because it was not fully logged...
INFO: Falling back to transitive mapping dependences for Copy Op 719 due to either imprecision in the runtime analysis or a bug.
INFO: Falling back to transitive mapping dependences for Inter Close Op 727 due to either imprecision in the runtime analysis or a bug.
Traceback (most recent call last):
  File "/nobackupp17/kbando/Softwares/legion/tools/legion_spy.py", line 13734, in <module>
    main(temp_dir)
  File "/nobackupp17/kbando/Softwares/legion/tools/legion_spy.py", line 13672, in main
    state.perform_logical_analysis(logical_checks, sanity_checks)
  File "/nobackupp17/kbando/Softwares/legion/tools/legion_spy.py", line 12792, in perform_logical_analysis
    if not task.perform_logical_dependence_analysis(perform_checks):
  File "/nobackupp17/kbando/Softwares/legion/tools/legion_spy.py", line 8611, in perform_logical_dependence_analysis
    if not op.perform_logical_analysis(init_fields, perform_checks):
  File "/nobackupp17/kbando/Softwares/legion/tools/legion_spy.py", line 6915, in perform_logical_analysis
    if not self.analyze_logical_requirement(idx, init_fields, perform_checks):
  File "/nobackupp17/kbando/Softwares/legion/tools/legion_spy.py", line 6787, in analyze_logical_requirement
    if not req.parent.perform_logical_analysis(0, path, self, req, field,
  File "/nobackupp17/kbando/Softwares/legion/tools/legion_spy.py", line 3272, in perform_logical_analysis
    self.logical_state[field].perform_logical_analysis(op, req, next_child,
  File "/nobackupp17/kbando/Softwares/legion/tools/legion_spy.py", line 3978, in perform_logical_analysis
    if not self.analyze_initial_close(close, op, req, perform_checks):
  File "/nobackupp17/kbando/Softwares/legion/tools/legion_spy.py", line 4617, in analyze_initial_close
    if op.mappings[req.index][self.field.fid].is_virtual():
TypeError: 'NoneType' object is not subscriptable
```
The full output and the logs are available at `/scratch2/bandokihiro/Issue1321`.

**Comment by lightsighter** [comment]
Can you attach the logs here so they don't get lost? This message though makes me a little concerned:

> Warning: skipping logical analysis of Attach Op 715 because it was not fully logged...

Usually that occurs when the logs are incomplete, and unlike Legion Prof, Legion Spy isn't designed to work in all cases if Legion doesn't shut down properly.

**Comment by bandokihiro** [comment]
[log_0.log](https://github.com/StanfordLegion/legion/files/9507538/log_0.log)
[log_1.log](https://github.com/StanfordLegion/legion/files/9507540/log_1.log)
[out.txt](https://github.com/StanfordLegion/legion/files/9507541/out.txt)

The run exited cleanly as far as I can tell.

**Comment by lightsighter** [comment]
Pull, rebuild legion, regenerate the logs, and then try again.

**Comment by bandokihiro** [comment]
It failed with the following last lines of output
```
INFO: Falling back to transitive mapping dependences for forward_euler_task due to either imprecision in the runtime analysis or a bug.
INFO: Falling back to transitive mapping dependences for Inter Close Op 721 due to either imprecision in the runtime analysis or a bug.
Warning: skipping logical analysis of Attach Op 743 because it was not fully logged...
INFO: Falling back to transitive mapping dependences for Copy Op 747 due to either imprecision in the runtime analysis or a bug.
INFO: Falling back to transitive mapping dependences for Inter Close Op 755 due to either imprecision in the runtime analysis or a bug.
Traceback (most recent call last):
  File "/nobackupp17/kbando/Softwares/legion/tools/legion_spy.py", line 13734, in <module>
    main(temp_dir)
  File "/nobackupp17/kbando/Softwares/legion/tools/legion_spy.py", line 13672, in main
    state.perform_logical_analysis(logical_checks, sanity_checks)
  File "/nobackupp17/kbando/Softwares/legion/tools/legion_spy.py", line 12792, in perform_logical_analysis
    if not task.perform_logical_dependence_analysis(perform_checks):
  File "/nobackupp17/kbando/Softwares/legion/tools/legion_spy.py", line 8611, in perform_logical_dependence_analysis
    if not op.perform_logical_analysis(init_fields, perform_checks):
  File "/nobackupp17/kbando/Softwares/legion/tools/legion_spy.py", line 6915, in perform_logical_analysis
    if not self.analyze_logical_requirement(idx, init_fields, perform_checks):
  File "/nobackupp17/kbando/Softwares/legion/tools/legion_spy.py", line 6787, in analyze_logical_requirement
    if not req.parent.perform_logical_analysis(0, path, self, req, field,
  File "/nobackupp17/kbando/Softwares/legion/tools/legion_spy.py", line 3272, in perform_logical_analysis
    self.logical_state[field].perform_logical_analysis(op, req, next_child,
  File "/nobackupp17/kbando/Softwares/legion/tools/legion_spy.py", line 3978, in perform_logical_analysis
    if not self.analyze_initial_close(close, op, req, perform_checks):
  File "/nobackupp17/kbando/Softwares/legion/tools/legion_spy.py", line 4617, in analyze_initial_close
    if op.mappings[req.index][self.field.fid].is_virtual():
TypeError: 'NoneType' object is not subscriptable
```
[log_0.log](https://github.com/StanfordLegion/legion/files/9579614/log_0.log)
[log_1.log](https://github.com/StanfordLegion/legion/files/9579615/log_1.log)
[out.txt](https://github.com/StanfordLegion/legion/files/9579617/out.txt)

**Comment by lightsighter** [comment]
Pull and try again.

**Comment by bandokihiro** [comment]
Fixed in d4f807d4, thanks.
